src/terminal/parser/OutputStateMachineEngine.cpp

```diff
@@ -905,23 +905,24 @@ bool OutputStateMachineEngine::_ParseHyperlink(const std::wstring_view string,
 {
     params.clear();
     uri.clear();
+    const auto parts = Utils::SplitString(string, L';');
+    if (parts.size() < 2)
+    {
+        return false;
+    }
+
     const auto len = string.size();
-    const size_t midPos = string.find(';');
-    if (midPos != std::wstring::npos)
+    if (len != 1)
     {
-        if (len != 1)
+        uri = til::at(parts, 1);
+        const auto paramStr = til::at(parts, 0);

```

Comment:
i'd like to understand what the len 1 check is doing now that we're using SplitString

Reply:
So the basic idea I think it's for a string `;`, there will be 2 parts which are all empty strings. I think it can be a short circuit logic to return early in this case. But you're right. It can be deleted to make this even cleaner

Reply:
@PankajBhojwani  Will it damage something if I remove the `len != 1` check? The main reason I leave it here is because I try to minimize my diffs.

Reply:
With the `string == L";"` check above we don't need the `len` check anymore!